### DISCUSSION TITLE
Add support for new host language

### DISCUSSION THREAD

**Discussion by wwall** [adrian-thurston/ragel#90] [General]
Good afternoon.
Please tell me the procedure for adding a new language for code generation - how to set up CMakeLists.txt and other files, how to add a new language to the assembly of the main ragel code

**Comment by oodler577** [comment]
@wwall Did you ever figure this out? I wish to do the same. What languages do you want to add support for? I want to add for Perl and maybe another language I am playing around with for 8 bit computers.

**Comment by wwall** [comment]
@oodler577 
on example of my lang (my lang looks like julia)
1. add lang support in colm (copy rlhc-julia.lm into new rlhc-lang.lm)
2. find word 'julia' in folder colm, add reference to 'lang'
3. compile colm
4. in ragel create folder host-lang, copy all files from host-julia, change in all files 'julia' to 'lang'
5. find word 'julia' in folder ragel add reference to 'lang'
6. compile ragel
